### PR TITLE
Feature/vdx 173 schema gen

### PR DESCRIPTION
**Pull request by sanderPostma**

### CODE REVIEW

File: packages/vc-handler-ld-local/src/types/types.ts [hunk 1]
```diff
@@ -218,14 +218,25 @@ export interface IVerifyPresentationLDArgs {
...
-  presentationPurpose?: typeof ProofPurpose
+  presentationPurpose?: IProofPurpose
 
   /**
    * Check status function, to check credentials that have a credentialStatus property
    */
   checkStatus?: Function
 }
 
+export interface IProofPurpose {
```
Comment (sanderPostma) [inline packages/vc-handler-ld-local/src/types/types.ts:229]: I could have made different types and or them like AuthenticationProofPurpose|  ControllerProofPurpose | AssertionProofPurpose, but I'm not so confident the API extractor will turn this into a correct OpenAPI v3 structure.

**Comment by sanderPostma** [comment]
Still not building consistently

File: packages/vc-handler-ld-local/src/types/types.ts (hunk 1 above)
Comment (nklomp) [reply packages/vc-handler-ld-local/src/types/types.ts:229]: Please do try that first though. The Proof Purposes are well known constructs in any SSI solution. This is really confusing to people that know SSI.

**Review by nklomp — CHANGES_REQUESTED** [review]
See remark. We really need different proof purposes, as these are core parts of any SSI solution. Current approach will confuse people that know SSI.
Also I would not see why it would go wrong, as anyofs  / inheritence is fully supported.

**Pull request closed without merging**